### DISCUSSION TITLE
LÖVE Potion 3.0 Prerelease 1

### DISCUSSION THREAD

**Discussion by TurtleP** [lovebrew/lovepotion#186] [Announcements]
This is the first Pre-Release for LÖVE Potion 3.0. It is based on LÖVE 12.0, which is currently unreleased.
You may view [its release notes here](https://github.com/love2d/love/blob/12.0-development/changes.txt#L1-L100).

## LÖVE 12 Related Changes

- Added support for launching a specific .lua file as the main file.
- Added love.parsedGameArguments and love.rawGameArguments tables, in the main thread.
- Removed `love.filesystem.newFile` (replaced by `love.filesystem.openFile`).
- Changed audio file decoding to choose the most appropriate decoder based on file contents instead of the file extension.
- Updated LuaSocket from 3.0-rc1 to 3.1.0.
- Added `Joystick:setPlayerIndex` and `Joystick:getPlayerIndex`.
- Added `Joystick:getGamepadType`.
- Added love.filesystem.mountFullPath and love.filesystem.unmountFullPath, including opt-in mount-for-write support.
- Added love.filesystem.mountCommonPath, unmountCommonPath, and getFullCommonPath.
- Added 'readonly' field to love.filesystem.getInfo's returned table.
- Added love.event.restart(optionalvalue). A new love.restart field will contain the value after restarting.
- Added love.system.getPreferredLocales.
- Added SoundData:copyFrom.
- Added SoundData:slice.
- Added optional stream type parameter to love.audio.newSource streaming sources ("file" or "memory"). It defaults to "file".

## Modules

- audio
- data
- event
- filesystem
- joystick
- love
- sound
- system
- thread
- timer
- touch

## Objects

- ByteData
- Channel
- CompressedData
- DataView
- Decoder
- File
- FileData
- Joystick
- SoundData
- Source
- Thread

### Wii U

- The binary must be placed in a specific path in order for it to start properly. This is intended behavior until `wut` can pass the current working directory properly to the application.
- `love.system.getPreferredLocales()` will always return an empty string. At the moment, I could not find a way to obtain the console's current language and region settings from `wut`.
- `love.system.getModel()` will always return an empty string. Currently there is no way to determine the model of the system, i.e. Basic or Deluxe.

---

There is now support for the Wii U, which the wuhb/rpx must be placed in a specific path: `sdmc:/wiiu/apps/LOVEPotion/` or it will not work properly. This is currently due to a limitation in wut since it does not give the launched directory to the application. This should hopefully be fixed in the future, since this also prevents unique game loading as a bundled application.

Please report any bugs to the GitHub issue tracker.

<hr /><em>This discussion was created from the release <a href='https://github.com/lovebrew/lovepotion/releases/tag/3.0-pre1'>3.0 Pre-Release 1</a>.</em>

**Comment by CarlosNunezMX** [comment]
You can use an system application to resolve the console region, as used by indexiine installer but using Wii U system menu like this
```c
#define BROWSER_PATH_EUR "dev:/sys/title/00050030/1001220a"
```

And using `dirExists` function to get console region
